test-app-reactive/src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstemplatepackagenameasync/service/MessageListener.kt

```diff
@@ -2,16 +2,21 @@ package uk.gov.justice.digital.hmpps.hmppstemplatepackagenameasync.service
 
 import com.fasterxml.jackson.databind.ObjectMapper
 import io.awspring.cloud.sqs.annotation.SqsListener
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.future.asCompletableFuture
+import kotlinx.coroutines.launch
 import org.springframework.stereotype.Service
+import java.util.concurrent.CompletableFuture
 
 @Service
 class InboundMessageListener(private val inboundMessageService: InboundMessageService, private val objectMapper: ObjectMapper) {
 
   @SqsListener(queueNames = ["inboundqueue"], factory = "hmppsQueueContainerFactoryProxy")
-  fun processMessage(rawMessage: String?) {
+  fun processMessage(rawMessage: String?): CompletableFuture<Unit> {
     val (Message) = objectMapper.readValue(rawMessage, Message::class.java)
     val event = objectMapper.readValue(Message, HmppsEvent::class.java)
-    inboundMessageService.handleMessage(event)
+    @Suppress("OPT_IN_USAGE")
+    return GlobalScope.launch { inboundMessageService.handleMessage(event) }.asCompletableFuture()
```

Comment:
slightly dubious about this, but needed a way to convert back from a `suspend` function to completable future so that we get the correct method signature